pkg/kubectl/cmd/set/set_serviceaccount.go

```diff
@@ -0,0 +1,212 @@
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+	"k8s.io/kubernetes/pkg/util/i18n"
+)
+
+var (
+	serviceaccount_resources = `
+	replicationcontroller (rc), deployment (deploy), daemonset (ds), job, replicaset (rs), statefulset`
+
+	serviceaccount_long = templates.LongDesc(`
+	Update ServiceAccount of pod template resources.
+
+	Possible resources (case insensitive) can be : 
+	` + serviceaccount_resources)
+
+	serviceaccount_example = templates.Examples(`
+	# Set ReplicationController nginx-controller's ServiceAccount to serviceaccount1
+	kubectl set serviceaccount replicationcontroller nginx-controller serviceaccount1
+
+	# Short form of the above command
+	kubectl set sa rc nginx-controller serviceaccount1
+
+	# Print result in yaml format of updated nginx controller from local file, without hitting apiserver
+	kubectl set sa -f nginx-rc.yaml serviceaccount1 --local --dry-run -o yaml
+	`)
+)
+
+type ServiceAccountConfig struct {
+	fileNameOptions resource.FilenameOptions
+
+	mapper                 meta.RESTMapper
+	encoder                runtime.Encoder
+	decoder                runtime.Decoder
+	out                    io.Writer
+	err                    io.Writer
+	dryRun                 bool
+	shortOutput            bool
+	all                    bool
+	record                 bool
+	output                 string
+	changeCause            string
+	local                  bool
+	categoryExpander       resource.CategoryExpander
+	clientMapper           resource.ClientMapper
+	typer                  runtime.ObjectTyper
+	namespace              string
+	enforceNamespace       bool
+	args                   []string
+	print                  func(obj runtime.Object) error
+	updatePodSpecForObject func(runtime.Object, func(*api.PodSpec) error) (bool, error)
+}
+
+func NewCmdServiceaccount(f cmdutil.Factory, out, err io.Writer) *cobra.Command {
+	saConfig := &ServiceAccountConfig{
+		out: out,
+		err: err,
+	}
+
+	cmd := &cobra.Command{
+		Use:     "serviceaccount (-f FILENAME | TYPE NAME) SERVICE_ACCOUNT",
+		Aliases: []string{"sa"},
+		Short:   i18n.T("Update ServiceAccount of a resource"),
+		Long:    serviceaccount_long,
+		Example: serviceaccount_example,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(saConfig.Complete(f, cmd, args))
+			if err := saConfig.Validate(); err != nil {
+				cmdutil.CheckErr(cmdutil.UsageError(cmd, err.Error()))
+			}
+			if err := saConfig.Run(); err != nil {
+				cmdutil.CheckErr(cmdutil.UsageError(cmd, err.Error()))
+			}
+		},
+	}
+	cmdutil.AddPrinterFlags(cmd)
+
+	usage := "identifying the resource to get from a server."
+	cmdutil.AddFilenameOptionFlags(cmd, &saConfig.fileNameOptions, usage)
+	cmd.Flags().BoolVar(&saConfig.all, "all", false, "select all resources in the namespace of the specified resource types")
+	cmd.Flags().BoolVar(&saConfig.local, "local", false, "If true, set image will NOT contact api-server but run locally.")
```

Comment:
I did not find any convention for flag descriptions , but we can go with capital letter.